docs/pages/repo/docs/core-concepts/caching/to-cache-or-not-to-cache.mdx

```diff
@@ -0,0 +1,51 @@
+---
+title: To Cache or Not to Cache
+description: Whether 'tis more performant to cache.
+---
+
+import Callout from "../../../../../components/Callout";
+
+# To Cache or Not To Cache
+
+In building a monorepo you need to decide on a per-task basis if you are able to skip the task based upon having run it previously. This behavior is controlled via `pipeline.<task>.cache` inside of `turbo.json`:
+
+```jsonc filename="/turbo.json"
+{
+  "$schema": "https://turbo.build/schema.json",
+  "pipeline": {
+    "lint": {
+      "cache": true
+    }
+  }
+}
+```
+
+Turborepo defaults to `"cache": true`, so you don't need to specify this value, meaning that this is an identical configuration:
+
+```jsonc filename="/turbo.json"
+{
+  "$schema": "https://turbo.build/schema.json",
+  "pipeline": {
+    "lint": {}
+  }
+}
+```
+
+In these examples we have told Turborepo to cache the `lint` tasks. But, since we've only specified the `cache` key, the only thing that is being cached is contents of `stdout`. For some types of tasks, such as `lint` or `test` this type of minimal configuration is likely correct!
+
+However, for most tasks you will also need to specify [_what_ to cache](./what-to-cache), as well as the [files](./file-configuration) and [environment variables](./environment-variables) that must match in order for the cache to be valid.
+
+## When Not To Cache
+
+<Callout type="info">
+  `"cache": false` does not mean "always run!" It simply means "if this task is to be executed it will not be restored from cache." Do not rely on `"cache": false` to trigger side effects like deployments.
+</Callout>
+
+Since caching is the default behavior, and is ideal for _most_ situations, knowing when to opt out of that behavior is important.
+
+1. **Tasks which execute extremely fast.** If you intend to use the remote cache and the task can be executed in less than the time of a network round trip, say, 100 milliseconds, you should consider not caching the task.
+2. **Tasks whose output assets are enormous.** If the consequence of running a task is the production of Docker Container it is possible that the time spent creating a cache artifact, uploading it, and downloading it, will exceed the amount of time to regenerate it.
```

Comment:
tiny sidenote fyi, we got an ask to help cache docker images recently https://github.com/vercel/turbo/issues/4932